packages/signers/src/turnkey-secp256k1-signer.ts

```diff
@@ -0,0 +1,124 @@
+import { TurnkeyClient } from "@turnkey/http";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { ethers } from "ethers"; // v6 helpers
+import 'dotenv/config';
+import type { Signer } from "@sovereign-sdk/signers";
+import { hexToBytes, bytesToHex } from "@sovereign-sdk/utils";
+import { Point } from "@noble/secp256k1";
+import { keccak_256 } from "@noble/hashes/sha3";
+
+// --- Turnkey Configuration ---
+const ORG_ID = process.env.TURNKEY_ORG_ID;
+const API_PUBLIC_KEY = process.env.TURNKEY_API_PUB;
+const API_PRIVATE_KEY = process.env.TURNKEY_API_PRIV;
+// This is the private key that will be used to sign the transaction.
+// It is managed by Turnkey and never leaves their environment.
+const ECDSA_KEY_ID = process.env.TURNKEY_ECDSA_KEY_ID; // your key ID
+
+if (!ORG_ID || !API_PUBLIC_KEY || !API_PRIVATE_KEY || !ECDSA_KEY_ID) {
+    throw new Error("Turnkey environment variables are not set. Please create a .env file.");
+}
+
+// Helper function to wait for a certain amount of time.
+const wait = (ms: number) => new Promise(r => setTimeout(r, ms));
+
+export class TurnkeySecp256k1Signer implements Signer {
+    private tk: TurnkeyClient;
+    public readonly _publicKey: Uint8Array;
+
+    private constructor(
+        publicKey: Uint8Array,
+        turnkeyClient: TurnkeyClient
+    ) {
+        this._publicKey = publicKey;
+        this.tk = turnkeyClient;
+
+        // Bind `this` to the methods to ensure the context is not lost
+        // when they are called by the Sovereign SDK.
+        this.sign = this.sign.bind(this);
+    }
+
+    public static async create(): Promise<TurnkeySecp256k1Signer> {
+        if (!/^[0-9a-fA-F]+$/.test(API_PRIVATE_KEY!)) {
```

Comment:
Lets remove these formatting checks, we won't always know the shape of their API keys and their API will return a error if they're malformed